pam/internal/adapter/gdmmodel.go

```diff
@@ -96,8 +96,14 @@ func (m *gdmModel) pollGdm() tea.Cmd {
 		})
 	}
 
-	if len(gdmPollResults) > 0 {
-		log.Debugf(context.TODO(), "Gdm Poll response is %v", gdmPollResults)
+	for _, result := range gdmPollResults {
+		// Don't log EventData_IsAuthenticatedRequested because it contains
+		// the user password
+		if result.GetIsAuthenticatedRequested() != nil {
+			log.Debugf(context.TODO(), "GDM poll returned: IsAuthenticatedRequested")
+		} else {
+			log.Debugf(context.TODO(), "GDM poll returned: %s", result.Data)
+		}
 	}
```

Comment:
This iteration must be done only when `log.IsLevelEnabled(log.DebugLevel)` is true or we end up with a no-op cycle for no reason.

Also, in case of `if result.GetIsAuthenticatedRequested() != nil` I'd like the logic to be a bit more complex:
 - If some extra pam parameter is provided (available only in debug builds and for developers only): show the password
 - Otherwise don't hide the whole message, but just obfuscate the challenge, or debugging what we received is just useless, as we may not catch issues we're meant to.

Reply:
> This iteration must be done only when log.IsLevelEnabled(log.DebugLevel) is true or we end up with a no-op cycle for no reason.

Agreed, we can optimize this by checking the log level before going into the for loop.

> Also, in case of `if result.GetIsAuthenticatedRequested() != nil` I'd like the logic to be a bit more complex:
> 
>     * If some extra pam parameter is provided (available only in debug builds and for developers only): show the password
> 
>     * Otherwise don't hide the whole message, but just obfuscate the challenge, or debugging what we received is just useless, as we may not catch issues we're meant to.

I doubt that it's worth the effort, and that it would be good enough to add a temporary debug log here in a development build if we ever find ourselves in the situation that we're not sure if the password is transmitted correctly in this request. And I think it's important to make sure that the password is never logged on a user's machine. If you think it's important to have more logging here, feel free to create a PR.